This is sort of messy... this doesn't fix all Windows builds, but does get the builds that worked before working again on my machine. Trying different builds shows that `clang-tblgen.exe` doesn't even understand this flag, while MSVC requires the flag for a different tblgen tool?

With the flag, at this PR or before https://github.com/google/llvm-bazel/pull/71 (?):

```
λ bazel build --config=msvc @llvm-project//...
INFO: Analyzed 1073 targets (3 packages loaded, 8422 targets configured).
INFO: Found 1073 targets...
ERROR: C:/users/scott/_bazel_scott/ebmakxjr/external/llvm-project/clang/BUILD:292:7: Generating code from table: lib/Sema/OpenCLBuiltins.td @llvm-project//clang:libsema_openclbuiltins_inc_gen__gen_clang_opencl_builtins_genrule failed (Exit 1)
clang-tblgen.exe: Unknown command line argument '--long-string-literals=0'.  Try: 'C:\users\scott\_bazel_scott\ebmakxjr\execroot\__main__\bazel-out\host\bin\external\llvm-project\clang\clang-tblgen.exe --help'
clang-tblgen.exe: Did you mean '--gen-clang-attr-list'?
ERROR: C:/users/scott/_bazel_scott/ebmakxjr/external/llvm-project/clang/unittests/BUILD:65:8 Generating code from table: lib/Sema/OpenCLBuiltins.td @llvm-project//clang:libsema_openclbuiltins_inc_gen__gen_clang_opencl_builtins_genrule failed (Exit 1)
INFO: Elapsed time: 1.821s, Critical Path: 0.31s
INFO: 0 processes.
FAILED: Build did NOT complete successfully
```

That one target without the flag:
```
λ bazel build --config=msvc @llvm-project//clang:libsema_openclbuiltins_inc_gen__gen_clang_opencl_builtins_genrule INFO: Analyzed target @llvm-project//clang:libsema_openclbuiltins_inc_gen__gen_clang_opencl_builtins_genrule (2 packages loaded, 505 targets configured).
INFO: Found 1 target...
Target @llvm-project//clang:libsema_openclbuiltins_inc_gen__gen_clang_opencl_builtins_genrule up-to-date:
  bazel-bin/external/llvm-project/clang/lib/Sema/OpenCLBuiltins.inc
INFO: Elapsed time: 1.281s, Critical Path: 0.01s
INFO: 0 processes.
INFO: Build completed successfully, 1 total action
```

Looking at a different target:

Without the flag:

```
λ bazel build --config=msvc @llvm-project//llvm:X86UtilsAndDesc
INFO: Analyzed target @llvm-project//llvm:X86UtilsAndDesc (0 packages loaded, 0 targets configured).
INFO: Found 1 target...
ERROR: C:/users/scott/_bazel_scott/ebmakxjr/external/llvm-project/llvm/BUILD:1680:16: C++ compilation of rule '@llvm-project//llvm:X86UtilsAndDesc' failed (Exit 2)
C:\users\scott\_bazel_scott\ebmakxjr\execroot\__main__\bazel-out\x64_windows-fastbuild\bin\external\llvm-project\llvm\_virtual_includes\X86CommonTableGen\X86GenInstrInfo.inc(37970): fatal error C1091: compiler limit: string exceeds 65535 bytes in length
Target @llvm-project//llvm:X86UtilsAndDesc failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 3.221s, Critical Path: 1.31s
INFO: 10 processes: 10 local.
FAILED: Build did NOT complete successfully
```

With the flag
```
λ bazel build --config=msvc @llvm-project//llvm:X86UtilsAndDesc
INFO: Analyzed target @llvm-project//llvm:X86UtilsAndDesc (1 packages loaded, 2765 targets configured).
INFO: Found 1 target...
Target @llvm-project//llvm:X86UtilsAndDesc up-to-date:
  bazel-bin/external/llvm-project/llvm/X86UtilsAndDesc.lib
INFO: Elapsed time: 11.850s, Critical Path: 10.52s
INFO: 48 processes: 48 local.
INFO: Build completed successfully, 53 total actions
```

Maybe revert https://github.com/google/llvm-bazel/pull/71 instead, and someone with more context could fix-forward on MSVC?

--- a/llvm-bazel/llvm-project-overlay/llvm/BUILD
+++ b/llvm-bazel/llvm-project-overlay/llvm/BUILD
@@ -1654,6 +1654,12 @@ gentbl(
         strip_include_prefix = "lib/Target/" + target["name"],
         tbl_outs = target["tbl_outs"],
         tblgen = ":llvm-tblgen",
+        # MSVC isn't happy with long string literals, while other compilers
+        # which support them get significant compile time improvements with
+        # them enabled. Ideally this flag would only be enabled on Windows via
+        # a select() on `@bazel_tools//src/conditions:windows,`, but that would
+        # require refactoring gentbl from a macro into a rule.
+        tblgen_args = "--long-string-literals=0",
         td_file = "lib/Target/" + target["name"] + "/" + target["short_name"] + ".td",
         td_srcs = [
             ":common_target_td_sources",

--- a/llvm-bazel/llvm-project-overlay/llvm/BUILD
+++ b/llvm-bazel/llvm-project-overlay/llvm/BUILD
@@ -1659,6 +1659,7 @@ gentbl(
         # them enabled. Ideally this flag would only be enabled on Windows via
         # a select() on `@bazel_tools//src/conditions:windows,`, but that would
         # require refactoring gentbl from a macro into a rule.
+        # TODO(#92): Refactor gentbl to support this use
         tblgen_args = "--long-string-literals=0",
         td_file = "lib/Target/" + target["name"] + "/" + target["short_name"] + ".td",
         td_srcs = [

--- a/llvm-bazel/llvm-project-overlay/llvm/tblgen.bzl
+++ b/llvm-bazel/llvm-project-overlay/llvm/tblgen.bzl
@@ -12,7 +12,15 @@ TODO(chandlerc): Currently this expresses include-based dependencies as
 correctly understood by the build system.
 """
 
-def gentbl(name, tblgen, td_file, td_srcs, tbl_outs, library = True, **kwargs):
+def gentbl(
+        name,
+        tblgen,
+        td_file,
+        td_srcs,
+        tbl_outs,
+        library = True,
+        tblgen_args = "",
+        **kwargs):
     """gentbl() generates tabular code from a table definition file.
 
     Args:
@@ -24,6 +32,7 @@ def gentbl(name, tblgen, td_file, td_srcs, tbl_outs, library = True, **kwargs):
         options passed to tblgen, and the out is the corresponding output file
         produced.
       library: Whether to bundle the generated files into a library.
+      tblgen_args: Extra arguments string to pass to the tblgen binary.
       **kwargs: Keyword arguments to pass to subsidiary cc_library() rule.
     """
     if td_file not in td_srcs:
@@ -36,30 +45,16 @@ def gentbl(name, tblgen, td_file, td_srcs, tbl_outs, library = True, **kwargs):
             outs = [out],
             tools = [tblgen],
             message = "Generating code from table: %s" % td_file,
-            # MSVC isn't happy with long string literals, while other compilers
-            # which support them get significant compile time improvements with
-            # them enabled.
-            # See https://github.com/google/llvm-bazel/pull/71 for context.
-            cmd = select({
-                "@bazel_tools//src/conditions:windows": (("$(location %s) -I external/llvm-project/llvm/include " +
-                                                          "-I external/llvm-project/clang/include " +
-                                                          "-I $$(dirname $(location %s)) " +
-                                                          "%s $(location %s) --long-string-literals=0 -o $@") % (
-                    tblgen,
-                    td_file,
-                    opts,
-                    td_file,
-                )),
-                "//conditions:default": (("$(location %s) -I external/llvm-project/llvm/include " +
-                                          "-I external/llvm-project/clang/include " +
-                                          "-I $$(dirname $(location %s)) " +
-                                          "%s $(location %s) -o $@") % (
-                    tblgen,
-                    td_file,
-                    opts,
-                    td_file,
-                )),
-            }),
+            cmd = (("$(location %s) -I external/llvm-project/llvm/include " +
+                    "-I external/llvm-project/clang/include " +
+                    "-I $$(dirname $(location %s)) " +
+                    "%s $(location %s) %s -o $@") % (
+                tblgen,
+                td_file,
+                opts,
+                td_file,
+                tblgen_args,
+            )),
         )
 
     # For now, all generated files can be assumed to comprise public interfaces.

